packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -38,127 +37,207 @@ const trace = makeTracer('VM');
+  const { updater: assetUpdater, notifier: assetNotifier } = makeNotifierKit(
     harden({
       compoundedInterest,
-      interestRate: shared.getInterestRate(),
+      interestRate: fixed.factoryPowers.getGovernedParams().getInterestRate(),
       latestInterestUpdate,
       totalDebt,
     }),
   );
 
+  /** @type {MutableState & ImmutableState} */
+  const state = {
+    ...fixed,
+    assetNotifier,
+    assetUpdater,
+    debtBrand: fixed.debtBrand,
+    vaultCounter: 0,
+    liquidationInProgress: false,
+    totalDebt,
+    compoundedInterest,
+    latestInterestUpdate,
+    outstandingQuote: null,
+  };
+
+  return state;
+};
+
+const helperBehavior = {
   /**
-   *
-   * @param {[key: string, vaultKit: InnerVault]} record
+   * @param {MethodContext} context
+   * @param {Amount<'nat'>} toMint
+   * @throws if minting would exceed total debt
    */
-  const liquidateAndRemove = ([key, vault]) => {
-    trace('liquidating', vault.getVaultSeat().getProposal());
-    liquidationInProgress = true;
+  checkDebtLimit: ({ state }, toMint) => {
+    const { factoryPowers, totalDebt } = state;
+    const debtPost = AmountMath.add(totalDebt, toMint);
+    const limit = factoryPowers.getGovernedParams().getDebtLimit();
+    if (AmountMath.isGTE(debtPost, limit)) {
+      assert.fail(X`Minting would exceed total debt limit ${q(limit)}`);
+    }
+  },
 
-    // Start liquidation (vaultState: LIQUIDATING)
-    return liquidate(
-      zcf,
-      vault,
-      debtMint.burnLosses,
-      liquidationStrategy,
-      collateralBrand,
-      penaltyPoolSeat,
-      loanParamGetters.getLiquidationPenalty(),
-    )
-      .then(() => {
-        prioritizedVaults?.removeVault(key);
-        liquidationInProgress = false;
-      })
-      .catch(e => {
-        liquidationInProgress = false;
-        // XXX should notify interested parties
-        console.error('liquidateAndRemove failed with', e);
-      });
-  };
+  /**
+   * @param {{ state: ImmutableState & MutableState, facets: * }} context FIXME
+   * @param {bigint} updateTime
+   * @param {ZCFSeat} poolIncrementSeat
+   */
+  chargeAllVaults: async (
+    { state, facets: { helper } },
+    updateTime,
+    poolIncrementSeat,
+  ) => {
+    trace('chargeAllVaults', { updateTime });
+    const interestRate = state.factoryPowers
+      .getGovernedParams()
+      .getInterestRate();
+
+    // Update state with the results of charging interest
+
+    const stateUpdates = chargeInterest(
+      {
+        mint: state.debtMint,
+        mintAndReallocateWithFee: state.factoryPowers.mintAndReallocate,
+        poolIncrementSeat,
+        seatAllocationKeyword: 'RUN',
+      },
+      {
+        interestRate,
+        chargingPeriod: state.factoryPowers
+          .getGovernedParams()
+          .getChargingPeriod(),
+        recordingPeriod: state.factoryPowers
+          .getGovernedParams()
+          .getRecordingPeriod(),
+      },
+      // TODO make something like _.pick
```

Comment:
Referring to https://lodash.com/docs/4.17.15#pick

Do you think that would be useful? If you do I'll make a ticket and link it here. Otherwise I'll just remove the comment.

Reply:
I agree that would sometimes be useful.